cmd/dependabot/internal/cmd/update.go

```diff
@@ -26,60 +26,81 @@ var (
 	inputServerPort int
 )
 
-var updateCmd = &cobra.Command{
-	Use:   "update [<package_manager> <repo> | -f <input.yml>] [flags]",
-	Short: "Perform an update job",
-	Example: heredoc.Doc(`
+func NewUpdateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update [<package_manager> <repo> | -f <input.yml>] [flags]",
+		Short: "Perform an update job",
+		Example: heredoc.Doc(`
 		    $ dependabot update go_modules rsc/quote
 		    $ dependabot update -f input.yml
 	    `),
-	RunE: func(cmd *cobra.Command, args []string) error {
-		var outFile *os.File
-		if output != "" {
-			var err error
-			outFile, err = os.Create(output)
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var outFile *os.File
+			if output != "" {
+				var err error
+				outFile, err = os.Create(output)
+				if err != nil {
+					return fmt.Errorf("failed to create output file: %w", err)
+				}
+				defer outFile.Close()
```

Comment:
In Go if a function returns an error you shouldn't touch the other variables returned because they're probably `nil`. 

Failing early here is intentional too because we want to tell the user right away that the output file they want can't be written.